### PR TITLE
Fix indexing in IntegralImageNormalEstimation #3573

### LINKED ISSUE #3573: Index out of bounds in IntegralImageNormalEstimation
## Your Environment

* Operating System and version: Windows 10 x64
* Compiler: MSVC 2017
* PCL Version: master

## Context

I try to do a normal estimation on a subset of points of a point cloud using IntegralImageNormalEstimation. When I do the computation on all points
it works but when I set the indices the app crashes. Note that the
depth dependen smoothing has to be disabled for the crash to happen.

## Possible Solution

There seems to be a mixup with the index that is used to index
into the output cloud in `IntegralImageNormalEstimation::computeFeaturePart`.
In the branch when `use_depth_dependent_smoothing_` is disabled
you can see at the end that `pt_index` is used instead of `idx` to index into the output cloud.
When `use_depth_dependent_smoothing_` is enabled the correct index is used.

I already have it fixed locally on my machine. If you want me to I can submit a pull request
although the fix is just renaming a varibale in two lines of code.

### PR DESCRIPTION
**Pull request by office-florian-hubner**
Fix wrong indexing in IntegralImageNormalEstimation

Fixes #3573

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Hey, I was wondering if it's possible to create a small test for this fix?

**Comment by office-florian-hubner** [comment]
Sure, but I won't have time before next week.

**Comment by kunaltyagi** [comment]
Shall I assign you an issue (missing test) and approve this for merge?

**Comment by office-florian-hubner** [comment]
I would just add the test to this PR but if it works better for you if we do it in a separate one it is also ok.

**Comment by kunaltyagi** [comment]
We'll wait for you in that case

**Comment by office-florian-hubner** [comment]
While writing the test case I found another bug in the IntegralImageNormalEstimation. In order to write the test case I also need to fix this.
Should we add this fix to this pull request?

File: features/include/pcl/features/impl/integral_image_normal.hpp [hunk 1]
```diff
@@ -1047,7 +1047,7 @@ pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::computeFeaturePart (con
           continue;
         }
 
-        if (u < border || v > right)
+        if (u < border || u > right)
```
Comment (kunaltyagi) [inline features/include/pcl/features/impl/integral_image_normal.hpp]: I wish there was a simpler way to write this check using `u` just once. No issues, just commenting

```cpp
abs(2*u - width) >= 2*(width - 2*border)
// or a function cal
```

File: features/include/pcl/features/impl/integral_image_normal.hpp [hunk 2]
```diff
@@ -1114,8 +1114,8 @@ pcl::IntegralImageNormalEstimation<PointInT, PointOutT>::computeFeaturePart (con
         }
         else
         {
-          output [pt_index].getNormalVector3fMap ().setConstant (bad_point);
-          output [pt_index].curvature = bad_point;
+          output [idx].getNormalVector3fMap ().setConstant (bad_point);
```
Comment (kunaltyagi) [inline features/include/pcl/features/impl/integral_image_normal.hpp]: Note to self: use `output.points.back()` in next patch for features

**Pull request merged**